addons/project_timesheet_holidays/tests/test_timesheet_global_time_off.py

```diff
@@ -215,14 +215,20 @@ def test_timesheet_creation_and_deletion_for_calendar_set_and_remove(self):
         # So we need to check that the timesheets don't have more than 8 hours per day.
         self.assertEqual(leave_task.effective_hours, 80)
 
-    def test_search_is_timeoff_task(self):
-        """ Test the search method on is_timeoff_task
-        with and without any hr.leave.type with timesheet_task_id defined"""
-        leaves_types_with_task_id = self.env['hr.leave.type'].search([('timesheet_task_id', '!=', False)])
-        self.env['project.task'].search([('is_timeoff_task', '!=', False)])
+    def test_timeoff_task_creation_with_global_leave(self):
+        """ Test the search method on is_timeoff_task"""
+        task_count = self.env['project.task'].search_count([('is_timeoff_task', '!=', False)])
 
-        leaves_types_with_task_id.write({'timesheet_task_id': False})
-        self.env['project.task'].search([('is_timeoff_task', '!=', False)])
+        # Create a leave and validate it
+        self.env['resource.calendar.leaves'].create({
+            'name': 'Test',
+            'calendar_id': self.test_company.resource_calendar_id.id,
+            'date_from': datetime(2021, 1, 4, 7, 0, 0, 0),
+            'date_to': datetime(2021, 1, 8, 18, 0, 0, 0),
+        })
+
+        new_task_count = self.env['project.task'].search_count([('is_timeoff_task', '!=', False)])
+        self.assertNotEqual(task_count, new_task_count)
```

Comment:
Instead of doing 'assertNotEqual', i'd suggest doing 
```py
assertEqual(task_count + 1, new_task_count)
```
This way, the test is a bit more precise, and it can detect a mistake like a duplication of the generated task.

Reply:
@damrOdoo Done :sparkles: